lib/common/common/src/validation.rs

```diff
@@ -142,6 +142,44 @@ pub fn validate_sha256_hash_option(value: &Option<impl AsRef<str>>) -> Result<()
         .unwrap_or(Ok(()))
 }
 
+pub fn validate_multi_vector<T>(multivec: &[Vec<T>]) -> Result<(), ValidationErrors> {
+    // non_empty
+    if multivec.is_empty() {
+        let mut errors = ValidationErrors::default();
+        let mut err = ValidationError::new("empty_multi_vector");
+        err.add_param(Cow::from("message"), &"multi vector must not be empty");
+        errors.add("data", err);
+        return Err(errors);
+    }
+
+    // check all individual vectors non-empty
+    if multivec.iter().any(|v| v.is_empty()) {
```

Comment:
It's just a small discussion, not a proposal. We can check only the first vector for non-empty because later we check that all vectors have the same dim. It will produce different error messages but we do less validation job. I think it doesn't worth it, but we can change this check.

Reply:
given they the number of vectors per multivector is rather low, I'd rather keep a clear error message instead of optimizing :)